src/sources/nats.rs

```diff
@@ -150,12 +151,9 @@ async fn nats_source(
         let mut stream = FramedRead::new(msg.data.as_ref(), decoder.clone());
         while let Some(next) = stream.next().await {
             match next {
-                Ok((events, _byte_size)) => {
+                Ok((events, byte_size)) => {
                     let count = events.len();
-                    emit!(OldEventsReceived {
-                        byte_size: events.size_of(),
-                        count
-                    });
+                    emit!(EventsReceived { count, byte_size });
```

Comment:
Is this `byte_size` the event byte size? `EventsReceived` appears to be emitting `component_received_event_bytes_total` which is intended to be the in memory size of the events.

Reply:
Honestly not sure 🤔 I can drop it back to the `events.size_of()` it was originally.

Reply:
Reverted in https://github.com/vectordotdev/vector/pull/14323/commits/ee4e00a32f771b8c9a770b76d50e0a2bd38f9781

Reply:
👍 We should be careful when adding and adjusting these metrics. It can be easy to make an error and we don't, as of yet, have great tests for asserting the reported values.